crates/core/src/client/message_handlers.rs

```diff
@@ -136,19 +142,36 @@ enum DecodedMessage<'a> {
 }
 
 impl DecodedMessage<'_> {
-    async fn handle(self, client: &ClientConnection) -> Result<(), MessageExecutionError> {
+    async fn handle(self, client: &ClientConnection, timer: Instant) -> Result<(), MessageExecutionError> {
+        let address = client.module.info().address;
         let res = match self {
             DecodedMessage::Call { reducer, args } => {
                 let res = client.call_reducer(reducer, args).await;
```

Comment:
Will this only return after all subscription messages have been sent out to clients? Or when exactly does this function return?

Reply:
Could we add a comment explaining when this method returns?

Reply:
this function `awaits` nestedly for `call_reducer_with_tx` https://github.com/clockworklabs/SpacetimeDB/blob/master/crates/core/src/host/wasm_common/module_host_actor.rs#L481, I think that's the function we know ends after finishing reducer and broadcasting for it.